### ISSUE TITLE
SFTP-based document ingestion

### ISSUE THREAD

**Issue by dontlaugh** [eikek/docspell#1868]
I have an rsync.net account, and it allows uploading/downloading files over SFTP.

It would be nice to schedule a periodic background job that does a one-way sync from my SFTP server into Docspell. To me this is similar to the scan mailboxes (over IMAP) feature, but for generic files on a remote server.

**Comment by eikek** [comment]
Thanks for the suggestion! It is true that this is very similar to the imap import, because you are probably not able to run something on the remote machine, like `dsc`? But I'm hesitant to add it to docspell itself, because there are just so many other options that would fit as well and I'm afraid of having too much to maintain. (my current thoughts on this anyways :)).

But I think this is a good use case for an addon. You can create an addon and let it run periodically. I did one to import my payment receipts that are on a closed platform. If you want to try this, [here](https://docspell.org/docs/addons/writing/) is some information to get started and there are some [examples](https://github.com/docspell), too. I'm happy to help, take any questions etc, but won't be able to do it myself anytime soon. I'm sure it is a useful addon for others to use. Addons could be integrated later into docspell properly, if it makes sense (they are used very often, for example).

**Comment by dontlaugh** [comment]
I control the machine docspell runs on, so I can indeed use `dsc` for this. I haven't actually played with dsc much.

From what I can tell from Writing Addons documentation, this looks like how I should proceed.

Full disclosure: I skipped this page in the docs because I'd assumed I'd need to write Scala! A language I'm not familiar with. From the looks of things, I can write addons in just about any language, and delegate much to `dsc`. Seems cool :+1:

**Comment by eikek** [comment]
Yeah, you can use whatever language you want, but Scala is nice of course 😏 In your case maybe it's just downloading files via scp/sftp and specifying them in the `newFiles` part of the [output](https://docspell.org/docs/addons/writing/#output).
